guides/keras_cv/retina_net_overview.py

```diff
@@ -0,0 +1,356 @@
+"""
+Title: Train an Object Detection Model on Pascal VOC 2007 using KerasCV
+Author: [lukewood](https://lukewood.xyz)
+Date created: 2022/08/22
+Last modified: 2022/08/22
+Description: Use KerasCV to train a RetinaNet on Pascal VOC 2007.
+"""
+
+"""
+## Overview
+
+KerasCV offers a complete set of APIs to train your own state-of-the-art,
+production-grade object detection model.  These APIs include object detection specific
+data augmentation techniques, models, and COCO metrics.
+
+To get started, let's sort out all of our imports and define global configuration parameters.
+"""
+
+import matplotlib.pyplot as plt
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow import keras
+from tensorflow.keras import optimizers
+
+import keras_cv
+from keras_cv import bounding_box
+import os
+
+BATCH_SIZE = 8
+EPOCHS = int(os.getenv("EPOCHS", "1"))
+CHECKPOINT_PATH = os.getenv("CHECKPOINT_PATH", "checkpoint/")
+INFERENCE_CHECKPOINT_PATH = os.getenv("INFERENCE_CHECKPOINT_PATH", CHECKPOINT_PATH)
+
+"""
+## Data loading
+
+In this guide, we use the function: `keras_cv.datasets.pascal_voc.load()` to load our
+data. KerasCV requires a `bounding_box_format` argument in all components that process
+bounding boxes.  To match the KerasCV API style, it is recommended that when writing a
+custom data loader, you also support a `bounding_box_format` argument.
+This makes it clear to those invoking your data loader what format the bounding boxes
+are in.
+
+For example:
+
+```python
+train_ds, ds_info = keras_cv.datasets.pascal_voc.load(
+    split='train', bounding_box_format='xywh', batch_size=8
+)
+```
+
+Clearly yields bounding boxes in the format `xywh`.  You can read more about
+KerasCV bounding box formats [in the API docs](https://keras.io/api/keras_cv/bounding_box/formats/).
+
+Our data comesloaded into the format
+`{"images": images, "bounding_boxes": bounding_boxes}`.  This format is supported in all
+KerasCV preprocessing components.
+
+Let's load some data and verify that our data looks as we expect it to.
+"""
+
+dataset, dataset_info = keras_cv.datasets.pascal_voc.load(
+    split="train", bounding_box_format="xywh", batch_size=9
+)
+
+
+def visualize_dataset(dataset, bounding_box_format):
+    color = tf.constant(((255.0, 0, 0),))
+    plt.figure(figsize=(10, 10))
+    for i, example in enumerate(dataset.take(9)):
+        images, boxes = example["images"], example["bounding_boxes"]
+        boxes = keras_cv.bounding_box.convert_format(
+            boxes, source=bounding_box_format, target="rel_yxyx", images=images
+        )
+        boxes = boxes.to_tensor(default_value=-1)
+        plotted_images = tf.image.draw_bounding_boxes(images, boxes[..., :4], color)
```

Comment:
`draw_bounding_boxes` doesn't have a way to control line thickness, does it? I found the boxes a bit hard to parse visually.

Reply:
it does not :/